### PR TITLE
Stream replacement

### PR DESCRIPTION
**Pull request by burner**
I wanted to use d-yaml, but then dmd told me it used std.stream. As I was the one to push for its deprecation I felled responsable to fix D-YAML.

This PR removes the use of std.stream. Currently, I created a tiny MemoryStream and File replacement.
Properly not mergeable right know, but a good start I think.

Update:
Removed some leftover code

### CODE REVIEW

File: source/dyaml/dumper.d [hunk 1]
```diff
@@ -59,7 +60,7 @@ import dyaml.tagdirective;
  * Write to memory:
  * --------------------
  * import std.stream;
- * auto stream = new MemoryStream();
```
Comment (gedaiu) [inline source/dyaml/dumper.d]: I think the best option is to use Ranges instead of streams...

Comment (gedaiu) [inline source/dyaml/dumper.d]: but this would work for now...

Comment (burner) [inline source/dyaml/dumper.d]: yes the solution is not pretty, but from what I saw introducing ranges into the library is a big task.

as you said, for now I think it is ok

**Comment by kiith-sa** [comment]
Thanks, merged; and sorry for the delay.

I'd like to redo the dumping code and also use ranges for the API... just need the time
which I still don't have.

**Comment by burner** [comment]
thanks for merging this

**Pull request merged**